### PR TITLE
feat(notifications): add email client

### PR DESCRIPTION
**Pull request by thevaibhav-dixit**

### CODE REVIEW

File: core/notifications/src/cli/mod.rs [hunk 1]
```diff
@@ -16,6 +16,8 @@ struct Cli {
     pg_con: String,
     #[clap(env = "MONGODB_CON")]
     mongodb_connection: Option<String>,
+    #[clap(env = "EMAIL_PASSWORD")]
+    email_password: Option<String>,
```
Comment (bodymindarts) [inline core/notifications/src/cli/mod.rs:20]: Why optional?

File: core/notifications/src/email_executor/config.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use serde::{Deserialize, Serialize};
+
+use super::lettre::LettreConfig;
+
+#[derive(Clone, Default, Debug, Deserialize, Serialize)]
+pub struct EmailExecutorConfig {
+    pub lettre: LettreConfig,
```
Comment (bodymindarts) [inline core/notifications/src/email_executor/config.rs:7]: Don’t want to expose internals like that. Change the smtp library should not be a breaking change

**Comment by vindard** [comment]
Rebasing to this PR once it's merged should help with the integration failure
https://github.com/GaloyMoney/galoy/pull/4004

File: core/notifications/src/email_executor/smtp/mod.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#[derive(Clone)]
+pub struct SmtpClient {
+    client: AsyncSmtpTransport<Tokio1Executor>,
+}
+
+impl SmtpClient {
+    pub fn init(config: SmtpConfig) -> Result<Self, SmtpError> {
+        let creds = Credentials::new(config.username, config.password);
+        let client: AsyncSmtpTransport<Tokio1Executor> =
+            AsyncSmtpTransport::<Tokio1Executor>::starttls_relay("smtp.gmail.com")?
```
Comment (bodymindarts) [inline core/notifications/src/email_executor/smtp/mod.rs:24]: Why is this endpoint hard coded?

Comment (thevaibhav-dixit) [reply core/notifications/src/email_executor/smtp/mod.rs:24]: was just for testing should come in via the config.

File: core/notifications/src/app/config.rs [hunk 4]
```diff
@@ -1,8 +1,9 @@
 use serde::{Deserialize, Serialize};
 
-use crate::push_executor::PushExecutorConfig;
+use crate::{email_executor::EmailExecutorConfig, push_executor::PushExecutorConfig};
 
 #[derive(Clone, Default, Serialize, Deserialize)]
 pub struct AppConfig {
     pub executor: PushExecutorConfig,
```
Comment (thevaibhav-dixit) [inline core/notifications/src/app/config.rs:7]: Note: rename this to push_executor

Comment (thevaibhav-dixit) [inline core/notifications/src/app/config.rs:7]: Note: rename this to push_executor

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**